pkg/pillar/zedUpload/ociutil/progress.go

```diff
@@ -0,0 +1,63 @@
+package ociutil
+
+import (
+	"io"
+)
+
+// ProgressWriter is a writer which will send the download progress
+type ProgressWriter struct {
+	w              io.Writer
+	updates        chan<- Update
+	size, complete int64
+}
+
+// Write write bytes and update the progress channel.
+// Returns number of bytes written and error, if any
+func (pw *ProgressWriter) Write(p []byte) (int, error) {
+	n, err := pw.w.Write(p)
+	if err != nil {
+		return n, err
+	}
+
+	pw.complete += int64(n)
+	// if our provides total size is 0, then we allow anything, so send a "Total"
+	// of the same amount as Complete
```

Comment:
Is this assuming that maxsize is zero?

If the controller doesn't set a maxsize, we should add some support (at the top of volumemgr somewhere)  where it sets maxsize based on the available disk space. Thus when we get to the actual download we should always have a maxsize (which might be a lot larger than the actual size). Hence my question about assumptions.

Reply:
See the previous comment. Let's move it down there.